### PR TITLE
Allow override of ingress apiVersion

### LINKED ISSUE #19578: [stable/drone] Uses too new apiVersion for ingress
**Describe the bug**

I pulled down the `stable/drone` chart and fed it a configuration with `ingress.enable=true`. This activates a template with this type information:

```
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
```

I know my version of kube is a little behind, but we should make the apiVersion overrideable. You can see info about the deprecation here: https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/ . Requiring folks to be on Kube 1.14 is probably smart but it would be helpful to override as is done in `role-binding-pipeline.yml`:

```
apiVersion: rbac.authorization.k8s.io/{{ required "A valid .Values.rbac.apiVersion entry required!" .Values.rbac.apiVersion }}
```

**Version of Helm and Kubernetes**:

I'm on helm `2.14.2` and kubernetes `1.13.10`

**Which chart**: `stable/drone`

**What happened**: Failed to install chart

**What you expected to happen**: It should have installed the chart with a valid ingress object

**How to reproduce it** (as minimally and precisely as possible): Use kube 1.13 and set `ingress.enabled=true`.

**Anything else we need to know**: Thanks for the nice chart!

### PR DESCRIPTION
**Pull request by davideagle**
Signed-off-by: Davíð Örn Jóhannsson @davideagle <davideaglephotos@gmail.com>

#### What this PR does / why we need it:
Allow override of ingress apiVersion For those of us that are still on older versions of k8s

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #19578 

#### Special notes for your reviewer:
@christian-roggia @paulczar

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/drone/templates/ingress.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {{- if .Values.ingress.enabled -}}
 {{- $fullName := include "drone.fullname" . }}
 {{- $httpPort := .Values.service.httpPort }}
-apiVersion: networking.k8s.io/v1beta1
+apiVersion: {{ default "networking.k8s.io/v1beta1" .Values.ingress.apiVersion }}
```
Comment (lachie83) [inline stable/drone/templates/ingress.yaml]: How about this to make it compatible with versions <= k8s 1.13

```suggestion
{{ if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" }}
apiVersion: networking.k8s.io/v1beta1
{{ else }}
apiVersion: extensions/v1beta1
{{ end -}}
```

File: stable/drone/values.yaml [hunk 2]
```diff
@@ -51,7 +51,10 @@ service:
 ingress:
   ## If true, Drone Ingress will be created.
   ##
-  enabled: false
```
Comment (davidkarlsen) [inline stable/drone/values.yaml:54]: should the default change?

Comment (davideagle) [reply stable/drone/values.yaml]: Makes sense to me - will make the change

Comment (davideagle) [reply stable/drone/values.yaml:54]: Reverted to default value

**Comment by zakkg3** [comment]
guys we are on K 1.17 already in a few months 1.15 will be out of support. what's the point of this?
you should focus on update your k8s instead of make the chart compatible with a going to be depreciated version.
Anyway if you like to run a compatible version feel free to install an older version of this chart. like 2.0

**Comment by zakkg3** [comment]
duplicated with https://github.com/helm/charts/pull/20591
conflicts not solved in almost a month
/hold

**Comment by christian-roggia** [comment]
This chart is now deprecated, please refer to https://github.com/drone/charts for further support and feature requests.

**Pull request closed without merging**